pkg/cmd/agent-task/capi/sessions.go

```diff
@@ -0,0 +1,208 @@
+			break
+		}
+		sessions = append(sessions, response.Sessions...)
+		page++
+	}
+
+	// Drop any above the limit
+	if len(sessions) > limit {
+		sessions = sessions[:limit]
+	}
+
+	// Hydrate the Sessions with pull request data.
+	Sessions, err := c.hydrateSessionPullRequests(sessions)
+	if err != nil {
+		return nil, err
+	}
+
+	return Sessions, nil
+}
+
+// hydrateSessionPullRequests hydrates pull request information in sessions
+func (c *CAPIClient) hydrateSessionPullRequests(sessions []session) ([]*Session, error) {
+	if len(sessions) == 0 {
+		return nil, nil
+	}
+
+	prNodeIds := make([]string, 0, len(sessions))
+
+	for _, session := range sessions {
+		prNodeID := generatePullRequestNodeID(int64(session.RepoID), session.ResourceID)
+		if slices.Contains(prNodeIds, prNodeID) {
+			continue
+		}
+		prNodeIds = append(prNodeIds, prNodeID)
+	}
+
+	apiClient := api.NewClientFromHTTP(c.httpClient)
+
+	var resp struct {
+		Nodes []struct {
+			PullRequest sessionPullRequest `graphql:"... on PullRequest"`
+		} `graphql:"nodes(ids: $ids)"`
+	}
+
+	host, _ := c.authCfg.DefaultHost()
+	err := apiClient.Query(host, "FetchPRs", &resp, map[string]any{
+		"ids": prNodeIds,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	prs := make([]*api.PullRequest, 0, len(prNodeIds))
+	for _, node := range resp.Nodes {
+		prs = append(prs, &api.PullRequest{
+			ID:             node.PullRequest.ID,
+			FullDatabaseID: node.PullRequest.FullDatabaseID,
+			Number:         node.PullRequest.Number,
+			Title:          node.PullRequest.Title,
+			State:          node.PullRequest.State,
+			URL:            node.PullRequest.URL,
+			Body:           node.PullRequest.Body,
+			CreatedAt:      node.PullRequest.CreatedAt,
+			UpdatedAt:      node.PullRequest.UpdatedAt,
+			ClosedAt:       node.PullRequest.ClosedAt,
+			MergedAt:       node.PullRequest.MergedAt,
+			Repository:     node.PullRequest.Repository,
+		})
+	}
+
+	newSessions := make([]*Session, 0, len(sessions))
+	// For each session, we need to attach the Pull Request
+	for _, s := range sessions {
+		// For each Pull Request, check if it matches the session
+		for _, pr := range prs {
+			if strconv.FormatInt(s.ResourceID, 10) == pr.FullDatabaseID {
+				newSessions = append(newSessions, &Session{
+					session:     s,
+					PullRequest: pr,
+				})
+			}
+		}
+	}
+
+	return newSessions, nil
+}
+
+// generatePullRequestNodeID converts an int64 databaseID and repoID to a GraphQL Node ID format
+// with the "PR_" prefix for pull requests
+func generatePullRequestNodeID(repoID, pullRequestID int64) string {
+	buf := bytes.Buffer{}
+	parts := []int64{0, repoID, pullRequestID}
+
+	encoder := msgpack.NewEncoder(&buf)
+	encoder.UseCompactInts(true)
+
+	// Encode the parts
+	err := encoder.Encode(parts)
+	if err != nil {
+		panic(err)
+	}
+
+	// Use URL-safe Base64 encoding without padding
+	encoded := base64.RawURLEncoding.EncodeToString(buf.Bytes())
+
+	// Return with the PR_ prefix
+	return "PR_" + encoded
```

Comment:
We don't need the comments since the code is clear enough.


```suggestion
	if err := encoder.Encode(parts); err != nil {
		panic(err)
	}

	encoded := base64.RawURLEncoding.EncodeToString(buf.Bytes())
	return "PR_" + encoded
```

Reply:
Fixed in 5281be467d5ab084e716b3c96fedc6a698013438